mantidproject/mantid
# Merge IndexPeaks and IndexPeaksWithSatellites. Deprecate IndexPeaksWithSatellites.

**Description of work.**

The process used to index peaks in `IndexPeaks` and `IndexPeaksWithSatellites` was the same and the code was very similar. The decision was made to extend `IndexPeaks` with the option to enter the modulation information and deprecate `IndexPeaksWithSatellites`. 

**To test:**

Run tests from
* #25660
* #22895 (if data from SNS available)

You can also try with [pr-26807.txt](https://github.com/mantidproject/mantid/files/3617004/pr-26807.txt) as the UB matrix and [unindexed-peaks.zip](https://github.com/mantidproject/mantid/files/3617009/unindexed-peaks.zip) to test this.

First run the following script and observe that it indexes 12 peaks and leaves the other 12 with HKL=[0,0,0] (unindexed).

```python
Load(Filename='unindexed-peaks.nxs', OutputWorkspace='sx_peaks')
LoadIsawUB(InputWorkspace='sx_peaks', Filename='pr-26807.txt')
IndexPeaks(PeaksWorkspace='sx_peaks', Tolerance=0.1, ToleranceForSatellite=0)
# IndexPeaks(PeaksWorkspace='sx_peaks', Tolerance=0.1, ToleranceForSatellite=0.1, ModVector1='0.333,0.333,0', ModVector2='-0.667,0.333,0', ModVector3='0.333,-0.667,0', MaxOrder=1)
```

Run the commented out `IndexPeaks` and check it now indexes additional peaks as satellites.

After the satellites are indexed run `SaveIsawUB` and check that the file contains a list of modulation vectors.

Fixes #26829
Fixes #26836 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
